update to ginkgo v2

Luap99 opened the pull request:
<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
none
```

edsantiago commented:
Not just rootless: debian-12 and fedora-37, root, are at 54 minutes right now. I expect they will time out.

edsantiago commented:
There's something horribly weird and broken in the new ginkgo: it does not seem to show command stderr. Example: [f37 root log](https://api.cirrus-ci.com/v1/task/6121147310276608/logs/main.log), search in-page for "`--filter takes no arg`". It shows up jumbled together in a block labeled "Timeline", but does not show up next to the actual command that generated it. This makes it impossible to diagnose actual errors. The relevant code, in `ginkgo/v2/reporters/default_reporter.go`, suggests that this is triggered when `--nodes > 1` but at first glance I don't see a way to fix it.

Obviously this renders the entire package useless, so there must be a way to override it, and I'm sure one of you lovely people will find the solution much more quickly than I can. I'll keep looking anyway though.

Luap99 commented:
@edsantiago There are changes how output now works, I am aware of this. Benchmarks are also broken.
This is not mergable like that, thus marked as WIP. For now I want to figure out the hang. I can repoduce locally, the problem is that we are leaking podman processes and the new ginkgo waits for them to be completed.

Example processes that hang:
```
pholzing  704177  0.1  0.1 1789984 41924 pts/4   Sl+  18:54   0:00 /home/pholzing/go/src/github.com/containers/podman/bin/podman --storage-opt vfs.imagestore=/tmp/imagecachedir --root /tmp/podman_test3608744664/root --runroot /tmp/podman_test3608744664/runroot --runtime crun --conmon /usr/bin/conmon --network-config-dir /home/pholzing/.config/cni/net.d --network-backend cni --cgroup-manager cgroupfs --tmpdir /tmp/podman_test3608744664 --events-backend file --db-backend boltdb --storage-driver vfs run --uts private quay.io/libpod/alpine:latest ls -l /proc/self/ns/uts
pholzing  704463  0.1  0.1 1863716 41976 pts/4   Sl+  18:54   0:00 /home/pholzing/go/src/github.com/containers/podman/bin/podman --storage-opt vfs.imagestore=/tmp/imagecachedir --root /tmp/podman_test2981800605/root --runroot /tmp/podman_test2981800605/runroot --runtime crun --conmon /usr/bin/conmon --network-config-dir /home/pholzing/.config/cni/net.d --network-backend cni --cgroup-manager cgroupfs --tmpdir /tmp/podman_test2981800605 --events-backend file --db-backend boltdb --storage-driver vfs run --pid private quay.io/libpod/alpine:latest ls -l /proc/self/ns/pid
pholzing  704662  0.1  0.1 1863460 40512 pts/4   Sl+  18:54   0:00 /home/pholzing/go/src/github.com/containers/podman/bin/podman --storage-opt vfs.imagestore=/tmp/imagecachedir --root /tmp/podman_test3588093402/root --runroot /tmp/podman_test3588093402/runroot --runtime crun --conmon /usr/bin/conmon --network-config-dir /home/pholzing/.config/cni/net.d --network-backend cni --cgroup-manager cgroupfs --tmpdir /tmp/podman_test3588093402 --events-backend file --db-backend boltdb --storage-driver vfs run --cgroupns private quay.io/libpod/alpine:latest ls -l /proc/self/ns/cgroup
```
As soon as I kill them manually ginkgo returns.
So now I need to figure out why this happens.

edsantiago commented:
```diff
diff --git a/test/e2e/containers_conf_test.go b/test/e2e/containers_conf_test.go
index 125a7bff1..72095d109 100644
--- a/test/e2e/containers_conf_test.go
+++ b/test/e2e/containers_conf_test.go
@@ -233,6 +233,7 @@ var _ = Describe("Verify podman containers.conf usage", func() {
 		Expect(hostNS).To(Equal(ctrNS))
 
 		session = podmanTest.Podman([]string{"run", option, "private", ALPINE, "ls", "-l", nspath})
+		Expect(session).Should(Exit(0))
 		fields = strings.Split(session.OutputToString(), " ")
 		ctrNS = fields[len(fields)-1]
 		Expect(hostNS).ToNot(Equal(ctrNS))
```

???

Luap99 commented:
That is a good catch but I don't it is the problem, the podman process is completely stucked and does not respond to any signal (well kill work obv) and it is not listed as zombie.

I can attach with a debugger and it looks like we are deadlocked on a container lock 
```
(dlv) bt
0  0x00007f977e506fd3 in ???
   at ?:-1
1  0x000000000047c2e0 in runtime.systemstack_switch
   at /usr/lib/golang/src/runtime/asm_amd64.s:459
2  0x0000000000415a05 in runtime.cgocall
   at /usr/lib/golang/src/runtime/cgocall.go:168
3  0x00000000010aedac in github.com/containers/podman/v4/libpod/lock/shm._Cfunc_lock_semaphore
   at _cgo_gotypes.go:227
4  0x00000000010affd8 in github.com/containers/podman/v4/libpod/lock/shm.(*SHMLocks).LockSemaphore
   at ./go/src/github.com/containers/podman/libpod/lock/shm/shm_lock.go:229
5  0x00000000010b0ea5 in github.com/containers/podman/v4/libpod/lock.(*SHMLock).Lock
   at ./go/src/github.com/containers/podman/libpod/lock/shm_lock_manager_linux.go:114
6  0x000000000123af8e in github.com/containers/podman/v4/libpod.(*Container).Kill
   at ./go/src/github.com/containers/podman/libpod/container_api.go:271
7  0x0000000001375b32 in github.com/containers/podman/v4/pkg/domain/infra/abi/terminal.ProxySignals.func1
   at ./go/src/github.com/containers/podman/pkg/domain/infra/abi/terminal/sigproxy_commn.go:36
8  0x000000000047e501 in runtime.goexit
   at /usr/lib/golang/src/runtime/asm_amd64.s:1594

```

But the thing is how the hell can it deadlock here, I have zero other podman processes running on the system which could be holding a lock, so what is holding the lock.

edsantiago commented:
Yeah... I noticed. Even with `WaitWithDefaultTimeout` it still hangs for me, unkillable except with -9. 

>zero other podman processes

Same here. But this is rootless... there's probably no pause process needed when running private namespaces, but could there be something that doesn't know that, and is waiting to join the pause process ns? (Ignore if this is a stupid comment. No need even to respond. Just thinking out loud.)

Luap99 commented:
Actually adding
```
session.WaitWithDefaultTimeout()
Expect(session).Should(Exit(0))
```
seems to fix it for me now. I only run containers_conf_test.go right now and before this change it hang every time now it passed three times in a row.

edsantiago commented:
Quick suggestion for next time you push: try changing the `.cirrus.yml` timeouts from 90m to 50 or even 45. Maybe not permanently, but it sure would save a lot of time in catching hangs.

vrothberg commented:
Thanks for looking into this, folks!

Luap99 commented:
Ok, hang is definitely fixed. Root cause is `podman rm --fa` (https://github.com/containers/podman/issues/18180) not working in all circumstances. Therefore it starts leaking processes and the ginkgo v2 doesn't like that. I assume the processes keep some std streams open which is tries to read.

Luap99 commented:
> Remote tests failing to compile: `undefined: GinkgoWriter`. But @Luap99 please don't re-push yet, I want some sample logs to download so I can fix logformatter. Should just be a few more minutes.

https://api.cirrus-ci.com/v1/artifact/task/5080120319279104/html/int-podman-fedora-37-root-container-boltdb.log.html

edsantiago commented:
Got em. (I need raw logs, the Cirrus ones. The logformatted ones are utter garbage right now, that's what I'm trying to fix)

Luap99 commented:
with the added `-p` instead of --nodes 3, `make localintegration` only takes 4-5 mins on my laptop (12 threads)

edsantiago commented:
Hey, are we expecting PANICs? In the process of updating logformatter, I see a bunch of these panics in `podman unshare` tests:
```
S [SKIPPED] [0.001 seconds]
         Podman unshare [BeforeEach]
           podman unshare
           Enter [AfterEach] Podman unshare - /var/tmp/go/src/github.com[/containers/podman/test/e2e/unshare_test.go:37](https://github.com/containers/podman/blob/de55faaecd839ec5f2bbc270ae27f37514de8ea8/test/e2e/unshare_test.go#L37) @ 04/13/23 13:04:41.586

           [PANICKED] Test Panicked
           In [AfterEach] at: /usr/lib/go-1.19/src/runtime/panic.go:260 @ 04/13/23 13:04:41.586
         
           runtime error: invalid memory address or nil pointer dereference
```
See [debian log](https://api.cirrus-ci.com/v1/task/6179631947055104/logs/main.log), search in-page for "PANIC". Compare to [current ginkgo log for a PR that merged today](https://api.cirrus-ci.com/v1/artifact/task/4530914328313856/html/int-podman-debian-12-root-host-boltdb.log.html).

The very weird thing is that the test shows as SKIPPED, and there is no link (at bottom) to the panic. No indication anywhere that there was a panic.

Anyhow, not my highest priority right now but someone should maybe look at it.

Luap99 commented:
Looks like ginkgo now still runs AfterEach() when we call Skip() in BeforeEach()

Luap99 commented:
> Looks like ginkgo now still runs AfterEach() when we call Skip() in BeforeEach()

Pushed a commit which should fix it, I think the new behavior makes sense. Otherwise cleanup was missing which caused us to leak tmp directories.

edsantiago commented:
I have a semi-working logformatter, would you like a copy? It makes reading logs much easier. (I was going to push it to your branch at the end of my day, but as long as you're still around, I can do so now if you want)

Luap99 commented:
I am done for today please push it.

edsantiago commented:
OK, if you're done for the day, I'll wait for the current test run to finish. That will give me better data (logs) from which I can maybe improve the script. Good night!

edsantiago commented:
Logformatter pushed and sort-of working. Some bugs still with in-page anchors and probably more I haven't found. Sample [remote](https://api.cirrus-ci.com/v1/artifact/task/6048850863980544/html/int-remote-fedora-37-root-host-boltdb.log.html) log shows the new format. I'm confused by the frequent "unable to connect to podman" errors in AfterEach when it fails. Can something be killing the server?

Some nice things about ginkgo v2 format:
* Retries are all done in the same block of the log.
* The Enter/Exit lines make it easy to gray out boring stuff

One not-so-nice thing:
* Flakes are no longer reported, making them very hard to detect

One thing we need to fix:
* Lots of "Captured Stdout/Stderr" blocks. These should be fixed in the code so the command is shown along with the output, in the proper sequence. (Not necessary in this PR. For later.)

Also, I screwed up on my push: dragged in the rest of `main`. I trust that's not a huge problem to fix.

HTH.

Luap99 commented:
>    Flakes are no longer reported, making them very hard to detect

I am pretty sure we can fix this manually. If you look at https://pkg.go.dev/github.com/onsi/ginkgo/v2/types#SpecReport we can get this info and the print it in any format we want similar to what we already do with the test timings at the bottom.

Luap99 commented:
Looks like flakes are still reported it just looks differently:
https://api.cirrus-ci.com/v1/artifact/task/6549968723378176/html/int-podman-fedora-36-rootless-host-boltdb.log.html#t--podman-run-net-container--copies-hosts-and-resolv--1
at the bottom you see flakes 3, search for `podman cp from ctr chown` which has the nasty `no logs from conmon` flake. The log cleary shows the retry: `Attempt #1 Failed.  Retrying ...`

edsantiago commented:
>at the bottom you see flakes 3,

Yes, I colorized that. The old ginkgo gave links to each flake; now it's necessary to grep for 'Attempt #' which of course also matches on the hard failures. Not a huge deal.

Luap99 commented:
Well I can make ginkgo print extra stuff if that makes your log formatter easier.
If you look at the AfterEach() nodes I can just read `NumAttempts` there from the SpecReport which tell us how often this was tried.

vrothberg commented:
> I have no idea how to figue out on what exactly it is waiting. It looks like it just hangs somewhere in the goruntime.

You can try send SIGABRT to the ginko process to force it dump the stack.  Maybe the stack trace is sufficiently long to get more hints.

Luap99 commented:
OK I think I fixed the network aardvark-dns leaks issue for the most part, the last networking issue I see locally is #17903.
I will add the `Serial` decorator and see if this truly fixes the issues once at for all.

Luap99 commented:
I will add the `Serial` decorator and see if this truly fixes the issues once at for all.

Yes this seems to work, no more `unknown FS magic on` in my local runs.

edsantiago commented:
So.... I'm working some more on logformatter and am noticing that logs have a LOT of nuls in them, in random places. I don't just mean the single nul in `cat /proc/self/attr/current`, I mean hundreds of nuls. Example: [f37 rootless sqlite](https://api.cirrus-ci.com/v1/task/6048164499685376/logs/main.log), search in-page for "`podman create --pod-id-file`" or "`regular capabilities`". There are many more.

logformatter already strips those, I added that because of the nasty nuls in the bottom line of each log. It's easy to just keep doing so. But..... why? Could this be a symptom of a larger problem? Is this something that should be investigated, or swept under the rug?

Luap99 commented:
> Could this be a symptom of a larger problem? Is this something that should be investigated, or swept under the rug?

I don't think this is a big deal but I take a look.

edsantiago commented:
One more thing: the `[SLOW TEST]` warning is gone. I don't think anyone other than me ever even looked at those anyway.

Luap99 commented:
> One more thing: the `[SLOW TEST]` warning is gone. I don't think anyone other than me ever even looked at those anyway.

From ginkgo v2.5 release notes:
> The --progress, --slow-spec-threshold, --always-emit-ginkgo-writer flags
and the SuppressProgressReporting decorator have all been deprecated.  Instead
the existing -v and -vv flags better capture the level of verbosity to display.  However,
a new --show-node-events flag is added to include node `> Enter` and `< Exit` events
in the spec timeline.

We still list the test timings at the bottom so we can use this to look for slow test if needed.

edsantiago commented:
@Luap99 my brain short-circuited while trying to re-push to your branch. So, should you so wish, please cherry-pick the latest logformatter from #18243 .

Luap99 commented:
I included the commit from #18243

Luap99 commented:
I split out my networking fixes into https://github.com/containers/podman/pull/18254 to make review and possible backports easier.

edsantiago commented:
Interesting new failure in [remote f37 root](https://api.cirrus-ci.com/v1/artifact/task/4979598220328960/html/int-remote-fedora-37-root-host-sqlite.log.html#t--add-volumes--1), in a CONTAINERS_CONF test:
```
Error: preparing container <ID> for attach: lsetxattr /tmp/podman_test2748995835/root/vfs-containers/<ID>/userdata/shm: operation not supported
```

Luap99 commented:
> `Error: preparing container <ID> for attach: lsetxattr /tmp/podman_test2748995835/root/vfs-containers/<ID>/userdata/shm: operation not supported`

Yup I broke it in `test/e2e: fix incorrect usage of CreateTempDirInTempDir() `, it tries to mount the full test tmpdir with :Z so it basically tired to relabel the container store + all the mount points there. Will be fixed in the next push.

Luap99 commented:
Third podman bug reported as https://github.com/containers/podman/issues/18259. This is hopefully the last one.

Luap99 commented:
I removed the WIP because I added a workaround for #18180.

@containers/podman-maintainers PTAL, I strongly recommend to look at this commit by commit.

Luap99 commented on .cirrus.yml:
```diff
@@ -574,7 +574,7 @@ local_integration_test_task: &local_integration_test_task
     matrix: *platform_axis
     gce_instance: *standardvm
-    timeout_in: 90m
+    timeout_in: 50m
```
@cevich @edsantiago Please let me know if the timeout change is ok for you.

Reason: Tests do not run that long and if there is a new bug which causes a hang we just wait for no good reason.

cevich replied:
I agree that 90m is probably too long.  But 50 seems a little short to me.  I remember not long ago, e2e tests taking 45min (wrong?).  There could easily be networking hiccups/slowdowns, and the set of tests is always expanding.  Could we do 55m as a compromise?

Luap99 replied:
I had 45 min for all the times I pushed this PR without issues, so I added a bit more margin to 50 mins. But lets wait for @edsantiago because he has the timing script and should know better what is realistic.

edsantiago commented:
I 100% support lowering the timeout. 45m is safe, I think, but we're likely to hit that within the next year so 50m is fine too.

| type | distro              | user     | DB     |    local |   remote | container |
| ---- | --------            | ----     | --     |    ----: |   -----: | --------: |
| int  | debian-12           | root     |        |    32:45 |    40:36 |
| int  | fedora-37           | root     |        |    40:34 |    40:35 |     39:06 |
| int  | fedora-38           | root     |        |    38:17 |    40:19 |     35:26 |
| int  | fedora-38           | root     | sqlite |    37:59 |    36:42 |
| int  | rawhide             | root     | sqlite |    37:00 |    41:25 |
| int  | debian-12           | rootless |        |    31:27 |          |
| int  | fedora-37           | rootless |        |    36:41 |          |
| int  | fedora-38           | rootless |        |    37:04 |          |
| int  | fedora-38           | rootless | sqlite |    35:58 |          |
| int  | rawhide             | rootless | sqlite |    36:52 |          |
| sys  | debian-12           | root     |        |    27:43 |    19:46 |
| sys  | fedora-37           | root     |        |    33:05 |    21:02 |
| sys  | fedora-38           | root     |        |    32:46 |    21:43 |
| sys  | fedora-38           | root     | sqlite |    30:44 |    21:10 |
| sys  | fedora-38-aarch64   | root     |        |    33:16 |    22:36 |
| sys  | rawhide             | root     | sqlite |    31:18 |    20:25 |
| sys  | debian-12           | rootless |        |    30:10 |          |
| sys  | fedora-37           | rootless |        |    30:26 |          |
| sys  | fedora-38           | rootless |        |    29:20 |    19:57 |
| sys  | fedora-38           | rootless | sqlite |    28:05 |          |
| sys  | rawhide             | rootless | sqlite |    30:23 |          |
| bud  | fedora-38           | root     |        |    30:54 |    31:12 |

I could probably write something to fetch timing results for the last 100 PRs and find the max, but I'm not sure it's worth the effort. In my months of looking at (selected) results, I don't remember seeing anything past 41-42.

podman-machine failures look real but I haven't delved too deeply into them.

edsantiago commented on test/e2e/common_test.go:
```diff
@@ -629,7 +629,12 @@ func (p *PodmanTestIntegration) Cleanup() {
+	// FIXME: Remove this special case when the issue is fixed.
+	// Special case rm -fa is not working correctly with dependencies, https://github.com/containers/podman/issues/18180
+	if !strings.Contains(rmall.ErrorToString(), "has dependent containers which must be removed before it") {
+		Expect(rmall).To(Exit(0), "command: %v\nstdout: %s\nstderr: %s", rmall.Command.Args, rmall.OutputToString(), rmall.ErrorToString())
```
Side note: this is going to trigger on the unlinkat/EBUSY flake.

vrothberg commented:
I'll grab a fresh coffee before reviewing :^)

edsantiago commented:
All `int` tests passed, with times of 26:44 to 38:24, but one timed out at 50m. Would presumably also time out at 90m. I don't understand why only one test would time out.

Luap99 commented:
I restarted it, my best guess is some flake leaving processes running...

Luap99 commented:
```
[+0490s]   Running: /var/tmp/go/src/github.com/containers/podman/bin/podman --root /tmp/podman_test1335685339/root --runroot /tmp/podman_test1335685339/runroot --runtime crun --conmon /usr/bin/conmon --network-config-dir /tmp/podman_test1335685339/root/etc/networks --network-backend netavark --cgroup-manager systemd --tmpdir /tmp/podman_test1335685339 --events-backend file --db-backend sqlite --storage-driver vfs ps -q
[+0490s]   Error: beginning transaction: database is locked
```

This is the only failure logged by ginkgo, passed on the retry. However the AfterEach cleanup still fired correctly and looks successful so I don't think it leaked a process.

edsantiago commented:
OMG.... CI is green. I'm still uncomfortable about that one timeout, and expect it to recur... but I am 100% OK with fixing that later if needed. @containers/podman-maintainers PTAL, this will potentially fix many flakes, and make debugging easier (better logs), and much more. (@Luap99 I'm still doing my final review, so, almost-lgtm but not quite yet).

edsantiago commented on Makefile:
```diff
@@ -558,8 +557,8 @@ test: localunit localintegration remoteintegration localsystem remotesystem  ##
 .PHONY: ginkgo-run
 ginkgo-run: .install.ginkgo
-	ACK_GINKGO_RC=true $(GINKGO) version
-	ACK_GINKGO_RC=true $(GINKGO) -v $(TESTFLAGS) -tags "$(TAGS) remote" $(GINKGOTIMEOUT) -cover -flakeAttempts 3 -progress -trace -noColor -nodes $(GINKGONODES) -debug $(GINKGOWHAT) $(HACK)
```
Removing `-cover` looks correct to me: it's handled in the `localunit` target above. I've traced the presence of `-cover` in this line all the way back to 2018, and believe it to be a copy/paste from the 2017 CRI-O import. But since there's a slight possibility that I'm wrong, I'm calling attention to it anyway. If anyone sees a reason to keep this, please speak now.

Luap99 commented:
> I'm still uncomfortable about that one timeout, and expect it to recur...

Yeah the unfortunate thing is this is really hard to track down. I currently test locally if the ginkgo timeout is helpful here, right now cirrus will loose some output on timeout. However if I can set a little bit lower timeout for ginkgo and it exits (not confirmed yet) then we can add a `ps auxww` to the cirrus script to check for running processes which should help debug what is hanging.

edsantiago commented:
>we can add a ps auxww to the cirrus script

AIUI that's not the way Cirrus works: the Cirrus timeout is a hard stop, kill everything, no grace period or chance for cleanup or handler. @cevich do you know if there's a soft timeout option, where an emergency handler could be run?

Luap99 commented:
> AIUI that's not the way Cirrus works: the Cirrus timeout is a hard stop, kill everything, no grace period or chance for cleanup or handler. @cevich do you know if there's a soft timeout option, where an emergency handler could be run?

I am well aware that is why I am talking about the ginkgo timeout: `ginkgo --timeout 45m ...`. In this case ginkgo should exit after that time assuming it is not effected by the hang as well. If ginkgo timeout 45 min and cirrus timeout 50 min then I have a small window afterwards to run debug commands.

edsantiago commented:
Oops! My bad, I didn't properly understand your use of "ginkgo timeout". Yes, using `ginkgo --timeout SHORTERVAL` sounds perfect (but for a later PR). Thanks for the clarification.

edsantiago commented on test/e2e/common_test.go:
```diff
@@ -167,11 +168,40 @@ func (p *PodmanTestIntegration) Setup() {
+},
 	func() {
-		sort.Sort(testResultsSortedLength{testResults})
+		testTimings := make(testResultsSorted, 0, 2000)
```
Thank you! I hope this finally fixes #8358

edsantiago reviewed (approved):
Final LGTM. Thanks so much for persevering.

rhatdan commented:
/lgtm

cevich commented on Makefile:
```diff
@@ -571,7 +570,7 @@ ginkgo-remote:
 .PHONY: testbindings
 testbindings: .install.ginkgo
-	ACK_GINKGO_RC=true $(GINKGO) -v $(TESTFLAGS) -tags "$(TAGS) remote" $(GINKGOTIMEOUT) -progress -trace -noColor -debug -timeout 30m  -v -r ./pkg/bindings/test
+	$(GINKGO) -v $(TESTFLAGS) --tags "$(TAGS) remote" $(GINKGOTIMEOUT) --trace --no-color --timeout 30m  -v -r ./pkg/bindings/test
```
There's a `$(GINKGOTIMEOUT)` specified here in addition to a `--timeout 30m`, not sure if that was intentional / doing some tricky thing with ginkgo or not.  Consider removing one or the other or adding a comment detailing why there are two timeout options.

cevich reviewed (approved):
LGTM - Two non-blocking `timeout` related comments that can be fixed later.

cevich commented on Makefile:
```diff
@@ -126,8 +126,7 @@ LIBSECCOMP_COMMIT := v2.3.3
 GINKGOTIMEOUT ?= -timeout=90m
```
Previously when there was a ginkgo hang, you could find out what it was by examining the ginkgo-note log files collected as CI-artifacts IIRC.  However, if Cirrus times out the task, the logs won't be collected.
